Stop server on SIGTERM by default

yawaramin opened the pull request:
Fix #174

yawaramin committed:
Stop server on SIGTERM by default

Fix #174

yawaramin committed:
Fix variable name

Willenbrink commented:
Looks good!

I was wondering if we should also handle SIGQUIT but that already works. See also [this discussion](https://stackoverflow.com/a/4047975) for the common signals to terminate processes. SIGHUP might be interesting at some point.

aantron commented on src/http/http.ml:
```diff
@@ -668,7 +668,10 @@ let serve_with_maybe_https
+let on_sigterm =
+  let promise, resolve = Lwt.wait () in
+  ignore (Lwt_unix.on_signal Sys.sigterm (fun _ -> Lwt.wakeup_later resolve ()));
+  promise
```
Creating this promise at program init time seems questionable now.

Before, the `never` promise was a sort of orphan promise in the heap that was fine to have floating around since init time.

But the `on_sigterm` promise should probably only be created if `serve`/`run` are actually called, since a program might end up with Dream linked in but never actually use it. It seems questionable to affect signal handling at init time even if Dream does not end up getting used (or at init time even if Dream does end up getting used).

yawaramin commented:
TODO: reconsider approach for multicore.

yawaramin committed:
Handle SIGTERM only inside Dream server

Don't install the signal handler at the module scope, to avoid polluting
the user's application with potentially unused handlers.

Willenbrink committed:
Handle SIGTERM and SIGINT

Willenbrink committed:
Update docs
